notes/dns-client-proxy.md

```diff
@@ -0,0 +1,667 @@
+address. 
+
+# TRUST ANCHOR OPTION
+
+~~~
+      +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+   0: |                          OPTION-CODE                          |
+      +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+   2: |                         OPTION-LENGTH                         |
+      +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+   4: |             ANCHORS-XML-LENGTH                                |
+      +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+   6: ~             ANCHORS-XML                                       ~
+      +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+   4: |             ANCHORS-P7S-LENGTH                                |
+      +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+   6: ~             ANCHORS-P7S                                       ~
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~
+
+where
+
+{newline="true"}
+OPTION-CODE
+
+: To be decided
+
+OPTION-LENGTH
+
+: Length of this option excluding the OPTION-CODE and OPTION-LENGTH fields
+
+ANCHORS-XML-LENGTH
+
+: Length of ANCHORS-XML in network byte order
+
+ANCHORS-XML
+
+: Trust anchors in XML format
+
+ANCHORS-P7S-LENGTH
+
+: Length of ANCHORS-P7S in network byte order
+
+ANCHORS-P7S
+
+: Signature in p7s format
+
+This option provides DNSSEC trust anchors as described in [@RFC7958].
+
+# Protocol Specification
+
+## Client Processing
+
+A stub resolver that wishes to use the PROXY CONTROL Option includes the
+option in all outgoing DNS requests that require privacy. The option
+should be initialized according to the needs of the application.
+In addition the PROXY SCOPE Option can be added. In requests, the Scope
+field is set to undefined.
+
+If the stub resolver receives a reply without a PROXY CONTROL Option
+included in the reply, then stub resolver has to assume that traffic will
+have Do53 levels of privacy.  Similarly, a lack of a PROXY SCOPE Option
+implies a global scope.
+
+If the stub resolver receives a BADPROXYPOLICY error then the proxy was
+unable to meet the requirements of the PROXY CONTROL Option.
+
+### Probing
+
+In cases where the stub resolver expects a local DNS proxy, or where the
+stub resolver has (a limited) fall back to more private transports, or
+when the security policy of the application is such that is better to fail
+than send queries over Do53, the stub resolver first sends a probing
+query to verify that the proxy supports the PROXY CONTROL and
+PROXY SCOPE Options.
+
+This request queries "resolver.arpa" for SOA records.
+The proxy MUST implement this as a
+Special Use Domain Name. The actual response is not important. The
+important part is that the proxy returns PROXY CONTROL and PROXY SCOPE
+Options as described in this document or sets
+the response code to BADPROXYPOLICY if it cannot meet specified policy.
```

Comment:
"or sets the response code" implies that in that case no PROXY CONTROL option is returned, but did we not initially have the idea that it would return such an option filled with the capabilities.
Also perhaps if quering for "resolver.arpa" SOA one could send an empty PROXY CONTROL option to just ask for a capability report?